File: src/MaksIT.CertsUI/Models/APIKeys/PatchApiKeyEntityScopeRequest.cs
using MaksIT.Core.Abstractions.Webapi;
using MaksIT.CertsUI.Engine;

namespace MaksIT.CertsUI.Models.APIKeys;

public class PatchApiKeyEntityScopeRequest : PatchRequestModelBase {
  public Guid? Id { get; set; }

  /// <summary>
  /// The ID of the entity (e.g., organization or application).
  /// </summary>
  public Guid EntityId { get; set; }

  /// <summary>
  /// The type of the entity (e.g., "Organization", "Application").
  /// </summary>
  public ScopeEntityType EntityType { get; set; }

  /// <summary>
  /// The scope granted to the user for this entity.
  /// </summary>
  public ScopePermission Scope { get; set; }
}

File: src/MaksIT.WebUI/src/models/certsUI/apiKeys/PatchApiKeyEntityScopeRequest.ts
import { PatchRequestModelBase } from '../../PatchRequestModelBase'
import { ScopeEntityType, ScopePermission } from '../../engine/scopeEnums'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.PatchApiKeyEntityScopeRequest</c>. */
export interface PatchApiKeyEntityScopeRequest extends PatchRequestModelBase {
  id?: string | null
  entityId: string
  entityType: ScopeEntityType
  scope: ScopePermission
}
